lib/pgBackRest/Expire.pm

```diff
@@ -224,24 +230,33 @@ sub process
         }
     }
 
-    # Default archive retention if not explicily set
-    if (defined($iFullRetention))
+    # if the archive retention is not explicitly set then determine what it should be set to
+    if (!defined($iArchiveRetention))
     {
-        if (!defined($iArchiveRetention))
+        if ($strArchiveRetentionType eq BACKUP_TYPE_FULL && defined($iFullRetention))
         {
             $iArchiveRetention = $iFullRetention;
         }
-
-        if (!defined($strArchiveRetentionType))
+        elsif ($strArchiveRetentionType eq BACKUP_TYPE_DIFF && defined($iDifferentialRetention))
         {
-            $strArchiveRetentionType = BACKUP_TYPE_FULL;
+            $iArchiveRetention = $iDifferentialRetention;
+        }
+        elsif ($strArchiveRetentionType eq BACKUP_TYPE_INCR)
+        {
+            confess &log(ERROR, &OPTION_RETENTION_ARCHIVE_TYPE . " (${strArchiveRetentionType}) not valid without option ".
+                         &OPTION_RETENTION_ARCHIVE, ERROR_OPTION_INVALID);
         }
     }
 
-    # If no archive retention type is set then exit
-    if (!defined($strArchiveRetentionType))
+&log (INFO, "retention-full: ".(defined($iFullRetention) ? $iFullRetention : 'UNDEF').
+            " retention-diff: ".(defined($iDifferentialRetention) ? $iDifferentialRetention : 'UNDEF').
+            " archive-type: ".(defined($strArchiveRetentionType) ? uc($strArchiveRetentionType) : 'UNDEF').
+            " retention-archive: ".(defined($iFullRetention) ? $iFullRetention : 'UNDEF'));
```

Comment:
Oh, crap, you weren’t supposed to see any of that. I forgot that my previous pull request would not be on that previous commit but rather the current state of the branch. My bad. Sorry, I was just testing everything out to verify the logic and then was going to decide what to put in ConfigLoad and what needed to stay in Expire.pm. I should have created a new branch…didn’t mean to waste your time...

Cynthia Shang
www.crunchydata.com





> On Aug 18, 2016, at 9:35 AM, David Steele <notifications@github.com> wrote:
> 
> In lib/pgBackRest/Expire.pm <https://github.com/pgbackrest/pgbackrest/pull/241#discussion_r75308225>:
> 
> >          }
> >      }
> >  
> > -    # If no archive retention type is set then exit
> > -    if (!defined($strArchiveRetentionType))
> > +&log (INFO, "retention-full: ".(defined($iFullRetention) ? $iFullRetention : 'UNDEF').
> > +            " retention-diff: ".(defined($iDifferentialRetention) ? $iDifferentialRetention : 'UNDEF').
> > +            " archive-type: ".(defined($strArchiveRetentionType) ? uc($strArchiveRetentionType) : 'UNDEF').
> > +            " retention-archive: ".(defined($iFullRetention) ? $iFullRetention : 'UNDEF'));
> You should convert this to a logDebugMisc call if it stays in. It takes care of all the pesky undef testing.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/pgbackrest/pgbackrest/pull/241/files/3720930ec746383201ca96997d2f11f44b5a5389..c5280871daeb45f76920a54d76869f61aaa0a888#r75308225>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ARXz_F7gKbTbXK2l8slG7A1rbiySFfikks5qhF-LgaJpZM4Jlwgi>.
>